### ISSUE TITLE
Google maps markers on iOS are much larger than on Android.

### ISSUE THREAD

**Issue by marcglasberg** [flutter/flutter#100997]
This issue is for https://pub.dev/packages/google_maps_flutter

I use some bitmap as a marker. It displays OK in Android. When I open the iOS app are many times larger. 

The workaround is creating different images of different sizes for each Platform, which is bad.

I found thin on StackOverflow: https://stackoverflow.com/questions/60935576/flutter-google-maps-markers-incorrect-size-on-ios

**Comment by darshankawar** [comment]
@marcglasberg 
Can you provide `flutter doctor -v`, plugin version and minimal code sample that shows the issue you are reporting ?

Thanks.

**Comment by sonelektrikci** [comment]
Situation is opposite for me. On Android it looks fine but on ios, it's small. I used this code to give size to marker: 

`
Future<Uint8List> getBytesFromAsset(String path, int width) async {
  ByteData data = await rootBundle.load(path);
  ui.Codec codec = await ui.instantiateImageCodec(data.buffer.asUint8List(),
      targetWidth: width);
  ui.FrameInfo fi = await codec.getNextFrame();
  return (await fi.image.toByteData(format: ui.ImageByteFormat.png))!
      .buffer
      .asUint8List();
}
`
Even if the given width same, I got different sized markers on android and ios.

**Comment by marcglasberg** [comment]
[√] Flutter (Channel stable, 2.10.4, on Microsoft Windows [Version 10.0.19044.1586], locale pt-BR)
    • Flutter version 2.10.4 at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c860cba910 (9 days ago), 2022-03-25 00:23:12 -0500
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Marcelo\AppData\Local\Android\sdk
    X cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1)
    • Android Studio at C:\Users\Marcelo\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.7199119
    • Flutter plugin version 57.0.1
    • Dart plugin version 201.9380
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Community Edition (version 2021.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.2
    • Flutter plugin version 65.2.4
    • Dart plugin version 213.7306

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.1586]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 99.0.4844.84
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 99.0.1150.55

[√] HTTP Host Availability                 
    • All required HTTP hosts are available

**Comment by marcglasberg** [comment]
![pin](https://user-images.githubusercontent.com/13332110/161407468-c854a0c1-1201-4b39-9888-19541f4a3108.png)

This is the image we're using as a marker.

**Comment by marcglasberg** [comment]
This is the code:

```
Marker(
      consumeTapEvents: true,
      markerId: MarkerId(marker.uid),
      position: marker.latLon,
      onTap: marker.onTap,
      icon: await BitmapDescriptor.fromAssetImage(
          ImageConfiguration.empty,
          assetName,
          package: Picture.defaultPackage),
      anchor: const Offset(0.5, 0.5),
      infoWindow: const InfoWindow(title: ""),
    );
```

**Comment by darshankawar** [comment]
@marcglasberg 
Thanks for the update. I think you are hitting same issue as https://github.com/flutter/flutter/issues/24865 which was closed in favor of https://github.com/flutter/flutter/issues/24213.
See the comment from team member: https://github.com/flutter/flutter/issues/24865#issuecomment-503253057

So I suggest you to follow-up in original issue for further updates.
Closing this as duplicate. If you disagree, write in comments and I'll reopen it.

**Comment by marcglasberg** [comment]
@darshankawar No, #24865 was not closed in favor of #24213. They are very different issues. One is about size differences for image markers, and the other is to allow widgets into the map. 

What happened is that a bot closed #24865 for no good reason at all. It's terrible to have important issues open for years, and then instead of fixing it a bot closes it as if it didn't exist. 

Please, if you can, reopen #24865. Or if you cannot reopened that issue anymore, I need to ask you to reopen this one.

This issue is not fixed and not the same as #24213.

Thank you!

**Comment by darshankawar** [comment]
Thanks for the feedback. Since https://github.com/flutter/flutter/issues/24865 is a very old issue, I will reopen this for further tracking.

**Comment by darshankawar** [comment]
@marcglasberg 
I tried your code snippet and used it in plugin's example and ran it on iOS simulator on which I get below result:

<img width="302" alt="Screenshot 2022-04-11 at 6 56 22 PM" src="https://user-images.githubusercontent.com/67046386/162749538-2ef9ff0e-c158-40f4-a144-a041a4851e39.png">

```
void _onMapCreated(GoogleMapController controllerParam) async {

    Marker(
        consumeTapEvents: true,
        markerId: MarkerId('place_name'),
        position: LatLng(9.669111, 80.014007),
      //  onTap: marker.onTap,
        icon: await BitmapDescriptor.fromAssetImage(
        ImageConfiguration.empty,
        'assets/3.0x/red_square.png',
        ),
    anchor: const Offset(0.5, 0.5),
    infoWindow: const InfoWindow(title: ""),
    );
    setState(() {
      controller = controllerParam;
    });
  }
```

Is this how you are seeing the marker too at your end ?

**Comment by marcglasberg** [comment]
@darshankawar No, it's not how I see it. I actually see the red circle correctly. It's just way larger than on Android.

**Comment by darshankawar** [comment]
Thanks for the update. Using below runnable code sample, I do see the icon as below which I used the one you provided earlier:

<img width="249" alt="Screenshot 2022-04-12 at 11 22 01 AM" src="https://user-images.githubusercontent.com/67046386/162890386-b576f163-5c46-44bd-abbf-5945f959e173.png">


<details>
<summary> runnable code sample </summary>

```

import 'dart:async';

import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

void main() {
  runApp(MaterialApp(
    home: HomePage(),
  ));
}

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  double _lat = 13.0827;
  double _lng = 80.2707;
  late final BitmapDescriptor myIcon;
  Completer<GoogleMapController> _controller = Completer();

  CameraPosition _currentPosition = CameraPosition(
    target: LatLng(13.0827, 80.2707),
    zoom: 12,
  );

  @override
  initState() {
    BitmapDescriptor.fromAssetImage(ImageConfiguration.empty, 'assets/icon.png', ).then((onValue) {
      myIcon = onValue;
    }
    );
    super.initState();
    _currentPosition = CameraPosition(
      target: LatLng(_lat, _lng),
      zoom: 12,
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Google Maps"),
      ),
      body: Container(
        height: double.infinity,
        width: double.infinity,
        child: GoogleMap(
          initialCameraPosition: _currentPosition,

          markers: {
            Marker(
              consumeTapEvents: true,
              markerId: MarkerId('current'),
              position: LatLng(_lat,_lng),
              icon: myIcon
            ),
          },
          onMapCreated: (GoogleMapController controller) {
            _controller.complete();
          },
        ),
      ),
    );
  }
}

```

</details>


Can you please try above code and check if you still get a bigger size of marker than what I am getting ?

**Comment by marcglasberg** [comment]
@darshankawar The icon I provided is a red circle: https://github.com/flutter/flutter/issues/100997#issuecomment-1086756352

In contrast, your image above shows the Google Maps location icon, which is not the image I provided.

**Comment by darshankawar** [comment]
@marcglasberg 
Thanks for the feedback. I looked at the current implementation again and observed that since you are using `BitmapDescriptor` to use a custom icon as marker, which leads me to this open issue which talks about same scenario as yours https://github.com/flutter/flutter/issues/34657. I suggest you to follow-up in the linked issue for further updates.
Closing this as a duplicate. If you disagree, write in comments and I'll reopen it.
